### PR TITLE
SEO-618 Revert "SEO-577: Mercury - show 404 on categories without mem…

### PR DESCRIPTION
**Pull request by gabrys**
…bers"

This reverts commit df494c2d7963ef45251a425d4285afb28ec8f53c.

### CODE REVIEW

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -360,19 +360,17 @@ public function getPage() {
 					case NS_CATEGORY:
```
Comment (WikiaTech) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: MAJOR Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* MAJOR 4 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MercuryApiController.class.php#L377](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L377): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. MAJOR [MercuryApiController.class.php#L379](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L379): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. MAJOR [MercuryApiController.class.php#L515](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L515): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MINOR [MercuryApiController.class.php#L10](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L10): Define a constant instead of duplicating this literal "title" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L316](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L316): Define a constant instead of duplicating this literal "article" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L353](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L353): Define a constant instead of duplicating this literal "details" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L358](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L358): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. MINOR [MercuryApiController.class.php#L533](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L533): Define a constant instead of duplicating this literal "languageName" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Comment by sebastianmarzjan** [comment]
LGTM, let's make the decision about next steps post-release

File: extensions/wikia/MercuryApi/MercuryApiController.class.php (hunk 1 above)
Comment (WikiaTech) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:360]: MAJOR Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* MAJOR 4 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MercuryApiController.class.php#L377](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L377): Reduce this "switch/case" number of lines from 16 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. MAJOR [MercuryApiController.class.php#L379](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L379): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. MAJOR [MercuryApiController.class.php#L515](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L515): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MINOR [MercuryApiController.class.php#L10](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L10): Define a constant instead of duplicating this literal "title" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L316](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L316): Define a constant instead of duplicating this literal "article" 5 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L353](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L353): Define a constant instead of duplicating this literal "details" 6 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. MINOR [MercuryApiController.class.php#L358](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L358): Replace this "switch" statement with "if" statements to increase readability. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. MINOR [MercuryApiController.class.php#L533](https://github.com/Wikia/app/blob/5dddbafa8fc9c0ddd23aec85eb0d7e7b9ffec764/extensions/wikia/MercuryApi/MercuryApiController.class.php#L533): Define a constant instead of duplicating this literal "languageName" 3 times. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

**Pull request merged**